### DISCUSSION TITLE
when using jinja template with filters that have boolean  values of type int , the filter not working

### DISCUSSION THREAD

**Discussion by moathOSA** [apache/superset#19279] [Q&A / Help]
when using a filter that have values like  0 , 1 which is a boolean filter 

when retrieving the values of this filter in jinja template i faced this error 

```
sequence item 0: expected str instance, int found
```

and this is how the filter look like
```
        {%- if (filter_values('cancelled') is defined) and filter_values('cancelled') -%}
            AND
            cancelled IN ({{ "'" + "','".join(filter_values('cancelled')) + "'" }})
        {%- endif -%}
```

superset version 1.4.1

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by villebro** [comment]
I'm not sure I consider this a bug. The problem is that the value that's coming back from `filter_values` is `int`, not `str`. You probably need to check the type of the value, and then omit the quotes and then cast to `str` in case of non-string filter values.

**Comment by moathOSA** [comment]
@villebro  so int values not supported in filters when using jinja template ?

**Comment by villebro** [comment]
No, this is basic python: the problem is you're calling `"'".join()` with non-string values.:
```python
>>> "'".join(["a", "b", "c"])
"a'b'c"
>>> "'".join(["a", "b", 1])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: sequence item 2: expected str instance, int found
```

**Comment by moathOSA** [comment]
@villebro  not using the join or the quotations ends up with 
a list  like this  cancelled IN ([0]) which lead to syntax error !

**Comment by villebro** [comment]
Again, this is not a bug in either Superset or Jinja, but something you need to address in how your jinja template code handles non string values. I'm going to convert this to a discussion; also, feel free to reach out on the Superset Slack on `#beginner-questions` or `#debug-together` where you may be able to get help from other users.

**Comment by villebro** [comment]
FYI @moathOSA , @betodealmeida is adding a feature here that will solve this problem: #19574

**Comment by jvelezmagic** [comment]
While waiting for the `where_in` macro, `( {{ filter_values('cancelled')|join(',') }} )` should also work.
